benchadapt/python/benchadapt/result.py

```diff
@@ -210,6 +250,58 @@ def to_publishable_dict(self):
         return res_dict
 
 
+def validate_or_remove_github_commit_key(res_dict: Dict, strict=False):
+    """
+    Mutate BenchmarkResult dictionary (result of asdict(self)) in-place to make
+    its `github` key property be compliant with the Conbench HTTP API:
+    - Remove it when it's set to `None` (silently)
+    - Remove it when it doesn't look good (but also emit a warning)
```

Comment:
IMO we should just error in this case (i.e. always make `strict=True`) because no commit information is a weird corner-case you should have to deliberately opt-in to; you don't want to end up like that because you configured your env vars shoddily.

That would also let this function be just validation, and the popping could be handled like other attributes in L238-248.

Reply:
I thought the same, see 
```
    # Note(JP): this is basically non-strict schema validation with the
    # following behavior: "ok, it looks like you wanted to send commit
    # information, but you didn't do it properly, and now I pretend as if
    # you really didn't want to do that, but I emit a bit of a warning".
    # We should make this stricter in the future.
```

and this was my attempt to not change everything at once

Reply:
now making this this stricter

Reply:
added commits: rewrote message once again. no more warnings, only errors. adjusted tests.